pkg/input/golden_test/tf/count-err/main.tf

```diff
@@ -0,0 +1,9 @@
+resource "aws_s3_bucket" "count_bucket" {
+  count         = var.create_bucket
+  bucket_prefix = "some_count"
+}
+
+resource "aws_s3_bucket" "for_each_bucket" {
+  for_each      = var.create_bucket ? {} : {"key": "value"}
+  bucket_prefix = "some_for_each"
+}
```

Comment:
1). `var.create_bucket` is probably an int / a number, given that we use it in the `count` above. Does 0 evaluate to `false`, and all other values are truthy, if used in this ternary? 
2) Why not also test the `for_each` here? If 1) is correct, then this ternary evaluates to `false`. Wouldn't it make sense to also exercise the `for_each` case?

Reply:
Oops, I used `var.create_bucket` both as a number as well as a boolean.  It should really only be one of both!  I will update the test case.

In this case, it doesn't really evaluate to `0/1/false/true` -- it fails evaluation because the variable is never declared.  This is intentional, as failing evaluation is what we're trying to reproduce in this test case.  So the ternary expression really evaluates to `null`.

I'm not sure what you mean by:

> Why not also test the for_each here?

This test case contains a resource for both `count` as well as `for_each` (though the name is confusing perhaps...)

Reply:
> I'm not sure what you mean by:
>> Why not also test the for_each here?
> This test case contains a resource for both count as well as for_each (though the name is confusing perhaps...)

I didn't know that the ternary expression evaluates to `null`, so the exact logic is irrelevant. My confusion came from the fact that the "true" states in the test cases were different, e.g. one would have evaluated to `{}` and the other one to `1`. But I saw that you switched them now as well so they evaluate to `{"key": "value"}` and `1` respectively. 

_Not that it matters_ as they'll both be null anyways...